lumeer-core/src/main/java/io/lumeer/core/facade/SearchFacade.java

```diff
@@ -60,17 +60,67 @@
    public List<Collection> searchCollections(Query query) {
       Set<Collection> collections = new HashSet<>();
 
-      if (!isCollectionSelectionOnly(query)) {
-         collections.addAll(getCollectionsByDocumentSearch(query));
+      if (query.getFulltext() != null && !query.getFulltext().isEmpty()) {
+         collections.addAll(getCollectionsByDocumentFulltext(query));
       }
 
-      collections.addAll(getCollectionsByCollectionSearch(query));
+      if(query.getDocumentIds() != null && !query.getDocumentIds().isEmpty()){
+         collections.addAll(getCollectionsByDocumentsIds(query.getDocumentIds()));
+      }
```

Comment:
I do not think it is the best idea to make separate requests for documentIDs and fulltext part of the query. The reason why we make another request based on `collectionCodes` is that collections are searched by this parameter in another database collection. But the ones searching for documents first should be merged into a single database query.

At the same time it would be worth making just document queries first and then get all collections (based on `collectionCodes` as well as retrieved documents) in a single database query. What do you think?

Reply:
Okay, that sounds good.

Reply:
But these two requests can't be marged. So I think It's good as it is.

Reply:
But merging these requests will lead to unnecessary when searching by documents ids..